src/AuthError.ts

```diff
@@ -1,17 +1,46 @@
-export class AuthError extends Error {
-  constructor(message: string, public code: AuthErrorCodes, public originError?: unknown | Error) {
-    super(message);
-  }
+import { ConnectJwt } from './types';
+
+export interface InvalidQshInfo {
+  computed: string;
+  received: string;
 }
 
-export enum AuthErrorCodes {
-  MISSED_TOKEN = 'MISSED_TOKEN',
+export enum AuthErrorCode {
+  MISSING_JWT = 'MISSING_JWT',
   UNAUTHORIZED_REQUEST = 'UNAUTHORIZED_REQUEST',
   FAILED_TO_DECODE = 'FAILED_TO_DECODE',
   UNKNOWN_ISSUER = 'UNKNOWN_ISSUER',
   WRONG_ISSUER = 'WRONG_ISSUER',
-  MISSED_QSH = 'MISSED_QSH',
+  MISSING_QSH = 'MISSING_QSH',
   INVALID_QSH = 'INVALID_QSH',
   INVALID_SIGNATURE = 'INVALID_SIGNATURE',
   TOKEN_EXPIRED = 'TOKEN_EXPIRED',
 }
+
+interface AuthErrorOptionalAttrs {
+  code: AuthErrorCode;
+  originError?: unknown | Error;
+  connectJwt?: ConnectJwt;
+  unverifiedConnectJwt?: ConnectJwt;
+  qshInfo?: InvalidQshInfo;
+}
+
+export class AuthError extends Error implements AuthErrorOptionalAttrs {
```

Comment:
I have added a lot of optional arguments and attributes to the error object so when it's logged it will contain all context available.
Today, for instance, when a token cannot be validated, we have to manually decrypt it without validation to obtain the issuer. It will be done by default now.